public/app/plugins/datasource/grafana-azure-monitor-datasource/components/AnalyticsConfig.tsx

```diff
@@ -140,28 +143,39 @@ export const AnalyticsConfig: FunctionComponent<Props> = (props: Props) => {
       <h3 className="page-heading">Azure Monitor Logs</h3>
       {credentialsEnabled && (
         <>
-          <Switch
-            label="Same details as Azure Monitor API"
-            checked={!logAnalyticsCredentials}
-            onChange={onLogAnalyticsSameAsChange}
-            {...tooltipAttribute}
-          />
+          {showLogAnalyticsCreds && (
+            <Switch
+              label="Same details as Azure Monitor API"
+              checked={!logAnalyticsCredentials}
+              onChange={onLogAnalyticsSameAsChange}
+              {...tooltipAttribute}
+            />
+          )}
           {showSameAsHelpMsg && (
             <div className="grafana-info-box m-t-2">
               <div className="alert-body">
                 <p>Re-enter your Azure Monitor Client Secret to use this setting.</p>
               </div>
             </div>
           )}
```

Comment:
Include this in the condition as well - we don't want to show this if the switch isn't visible.

It might not in practice, but we want to be extra sure 😅